docs/adrs/0549-composite-run-steps.md

```diff
@@ -0,0 +1,321 @@
+    - id: step1
+      uses: actions/setup-python@v1
+    - id: step2
+      uses: actions/setup-node@v2
+    - uses: actions/checkout@v2
+    - uses: user/composite@v1
+    - name: workflow step 1
+      run: echo hello world 3
+    - name: workflow step 2
+      run: echo hello world 4
+```
+
+Example `user/composite/action.yml`
+
+```yaml
+runs:
+  using: "composite"
+  steps:
+    - run: pip install -r requirements.txt
+    - run: npm install
+```
+
+Example Output
+
+```yaml
+[npm installation output]
+[pip requirements output]
+echo hello world 3
+echo hello world 4
+```
+
+We add a token called "composite" which allows our Runner code to process composite actions. By invoking "using: composite", our Runner code then processes the "steps" attribute, converts this template code to a list of steps, and finally runs each run step sequentially. If any step fails and there are no `if` conditions defined, the whole composite action job fails. 
+
+## Inputs
+
+Example `workflow.yml`:
+
+```yaml
+steps: 
+  - id: foo
+    uses: user/composite@v1
+    with:
+      your_name: "Octocat"
+```
+
+Example `user/composite/action.yml`:
+
+```yaml
+inputs:
+  your_name:
+    description: 'Your name'
+    default: 'Ethan'
+runs:
+  using: "composite"
+  steps: 
+    - run: echo hello ${{ inputs.your_name }}
+```
+
+Example Output:
+
+```
+hello Octocat
+```
+
+Each input variable in the composite action is only viewable in its own scope.
+
+## Outputs
+
+Example `workflow.yml`:
+
+```yaml
+...
+steps: 
+  - id: foo
+    uses: user/composite@v1
+  - run: echo random-number ${{ steps.foo.outputs.random-number }} 
+```
+
+Example `user/composite/action.yml`:
+
+```yaml
+outputs:
+  random-number: 
+    description: "random number"
+runs:
+  using: "composite"
+  steps: 
+    - run: echo "::set-output name=random-number::$(echo $RANDOM)"
+```
+
+Example Output:
+
+```
+::set-output name=my-output::43243
+random-number 43243
+```
+
+Each of the output variables from the composite action is viewable from the workflow file that uses the composite action. In other words, every child action output(s) is viewable only by its parent using dot notation (ex `steps.foo.outputs.bar`).
+
+Moreover, the output ids are only accessible within the scope where it was defined. Note that in the example above, in our `workflow.yml` file, it should not have access to output id (i.e. `my-output`). For example, in the `workflow.yml` file, you can't run `foo.steps.my-step.my-output`.
+
+## Context
+
+Similar to the workflow file, the composite action has access to the [same context objects](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts) (ex: `github`, `env`, `strategy`). 
+
+## Environment
+
+Example `workflow.yml`:
+
+```yaml
+env:
+  NAME1: test1
+  SERVER: production
+steps: 
+  - id: foo
+    uses: user/test@v1
+  - run: echo Server $SERVER
+```
+
+Example `user/composite/action.yml`:
+
+```yaml
+using: 'composite' 
+env:
```

Comment:
Will `env` be top level, or nested under `runs`, as it is when writing a docker action?

Does the behaviour also align with how `env` works for docker actions? (where the workflow can't override `runs.env`)